web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Drupal\self_deposit\Form;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\webform\Element\WebformAjaxElementTrait;
+use Drupal\webform\Plugin\WebformHandlerMessageInterface;
+use Drupal\webform\WebformSubmissionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\node\Entity\Node;
+use Drupal\paragraphs\Entity\Paragraph;
+
+/**
+ * Defines a webform that resends webform submission.
+ */
+class SelfDepositCreateForm extends FormBase {
+
+  use WebformAjaxElementTrait;
+
+  /**
+   * A webform submission.
+   *
+   * @var \Drupal\webform\WebformSubmissionInterface
+   */
+  protected $webformSubmission;
+
+  /**
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * @var \Drupal\webform\WebformSubmissionConditionsValidatorInterface
+   */
+  protected $conditionsValidator;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\asu_deposit_methods\DepositUtils
+   */
+  protected $depositUtils;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'self_deposit_create';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    $instance = parent::create($container);
+    $instance->configFactory = $container->get('config.factory');
+    $instance->conditionsValidator = $container->get('webform_submission.conditions_validator');
+    $instance->entityTypeManager = $container->get('entity_type.manager');
+    $instance->depositUtils = $container->get('asu_deposit_methods.deposit_utils');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, WebformSubmissionInterface $webform_submission = NULL) {
+    $this->webformSubmission = $webform_submission;
+    // Apply variants to the webform.
+    $webform = $webform_submission->getWebform();
+    $webform->applyVariants($webform_submission);
+
+    $form['actions']['submit'] = [
+        '#type' => 'submit',
+        '#value' => 'Create Repository Item',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->createItem($this->webformSubmission);
+    $this->messenger()->addStatus('Successfuly created repository item.');
```

Comment:
Can we get this to return a link to the new item node?

Reply:
Addressed in latest commit